apps/meteor/client/views/admin/ABAC/RoomAttributesContextualBar.tsx

```diff
@@ -0,0 +1,102 @@
+import { ContextualbarTitle } from '@rocket.chat/fuselage';
+import { useEffectEvent } from '@rocket.chat/fuselage-hooks';
+import { useEndpoint, useRouteParameter, useToastMessageDispatch } from '@rocket.chat/ui-contexts';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { FormProvider, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+import type { AdminABACRoomAttributesFormFormData } from './AdminABACRoomAttributesForm';
+import AdminABACRoomAttributesForm from './AdminABACRoomAttributesForm';
+import { ContextualbarClose, ContextualbarHeader } from '../../../components/Contextualbar';
+import { ABACQueryKeys } from '../../../lib/queryKeys';
+
+type RoomAttributesContextualBarProps = {
+	attributeId?: string;
+	attributeData?: {
+		key: string;
+		values: string[];
+		usage: Record<string, boolean>;
+	};
+	onClose: () => void;
+};
+
+const RoomAttributesContextualBar = ({ attributeData, onClose }: RoomAttributesContextualBarProps) => {
+	const { t } = useTranslation();
+	const queryClient = useQueryClient();
+
+	const methods = useForm<{
+		name: string;
+		attributeValues: { value: string }[];
+		lockedAttributes: { value: string }[];
+	}>({
+		defaultValues: attributeData
+			? {
+					name: attributeData.key,
+					attributeValues: [{ value: '' }],
+					lockedAttributes: attributeData.values.map((value) => ({ value })),
+				}
+			: {
+					name: '',
+					attributeValues: [{ value: '' }],
+					lockedAttributes: [],
+				},
+	});
+
+	const { getValues } = methods;
+
+	const attributeId = useRouteParameter('id');
+	const createAttribute = useEndpoint('POST', '/v1/abac/attributes');
+	const updateAttribute = useEndpoint('PUT', '/v1/abac/attributes/:_id', {
+		_id: attributeId ?? '',
+	});
+
+	const onSave = useEffectEvent(async (data: AdminABACRoomAttributesFormFormData) => {
+		const payload = {
+			key: data.name,
+			values: [...data.lockedAttributes.map((attribute) => attribute.value), ...data.attributeValues.map((attribute) => attribute.value)],
+		};
+		if (attributeId) {
+			await updateAttribute(payload);
+		} else {
+			await createAttribute(payload);
+		}
+	});
+
+	const dispatchToastMessage = useToastMessageDispatch();
+
+	const saveMutation = useMutation({
+		mutationFn: onSave,
```

Comment:
Prefer direct function literal declaration to avoid type mismatch.

```suggestion
		mutationFn: async (data: AdminABACRoomAttributesFormFormData) => {
				const payload = {
					key: data.name,
					values: [...data.lockedAttributes.map((attribute) => attribute.value), ...data.attributeValues.map((attribute) => attribute.value)],
				};
				if (attributeId) {
					await updateAttribute(payload);
				} else {
					await createAttribute(payload);
				}
			},
```